genesyscloud/architect_emergencygroup/resource_genesyscloud_architect_emergencygroup.go

```diff
@@ -0,0 +1,193 @@
+		Name:               &name,
+		Enabled:            &enabled,
+		EmergencyCallFlows: buildSdkEmergencyGroupCallFlows(d),
+	}
+
+	// Optional attributes
+	if description != "" {
+		emergencyGroup.Description = &description
+	}
+
+	if divisionId != "" {
+		emergencyGroup.Division = &platformclientv2.Writabledivision{Id: &divisionId}
+	}
+
+	log.Printf("Creating emergency group %s", name)
+	eGroup, _, err := ap.createArchitectEmergencyGroup(ctx, emergencyGroup)
+	if err != nil {
+		return diag.Errorf("Failed to create emergency group %s: %s", name, err)
+	}
+
+	d.SetId(*eGroup.Id)
+
+	log.Printf("Created emergency group %s %s", name, *eGroup.Id)
+
+	return readEmergencyGroup(ctx, d, meta)
+}
+
+func readEmergencyGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	ap := getArchitectEmergencyGroupProxy(sdkConfig)
+
+	log.Printf("Reading emergency group %s", d.Id())
+	return genesyscloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		emergencyGroup, resp, getErr := ap.getAllArchitectEmergencyGroup(ctx, d.Id())
+		if getErr != nil {
+			if genesyscloud.IsStatus404(resp) {
+				return retry.RetryableError(fmt.Errorf("Failed to read emergency group %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read emergency group %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceArchitectEmergencyGroup())
+
+		if emergencyGroup.State != nil && *emergencyGroup.State == "deleted" {
+			d.SetId("")
+			return nil
+		}
+
+		d.Set("name", *emergencyGroup.Name)
+		d.Set("division_id", *emergencyGroup.Division.Id)
+
+		if emergencyGroup.Description != nil {
+			d.Set("description", *emergencyGroup.Description)
+		} else {
+			d.Set("description", nil)
+		}
+
+		if emergencyGroup.Enabled != nil {
+			d.Set("enabled", *emergencyGroup.Enabled)
+		} else {
+			d.Set("enabled", nil)
+		}
+
+		if emergencyGroup.EmergencyCallFlows != nil && len(*emergencyGroup.EmergencyCallFlows) > 0 {
+			d.Set("emergency_call_flows", flattenEmergencyCallFlows(*emergencyGroup.EmergencyCallFlows))
+		} else {
+			d.Set("emergency_call_flows", nil)
+		}
+
+		log.Printf("Read emergency group %s %s", d.Id(), *emergencyGroup.Name)
+		return cc.CheckState()
+	})
+}
+
+func updateEmergencyGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	divisionId := d.Get("division_id").(string)
+	enabled := d.Get("enabled").(bool)
+
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	ap := getArchitectEmergencyGroupProxy(sdkConfig)
+
+	diagErr := genesyscloud.RetryWhen(genesyscloud.IsVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+		// Get current emergency group version
+		emergencyGroup, resp, getErr := ap.getAllArchitectEmergencyGroup(ctx, d.Id())
```

Comment:
It's a bit confusing how the function "getAllArchitectEmergencyGroup" returns one emergency group by ID. Was this a mistake, or is it how the API works?

Reply:
No.  I must have cut and past then forgot to remove the All.  Fixed